topydo/lib/PrettyPrinterFilter.py

```diff
@@ -57,15 +60,21 @@ def filter(self, p_todo_str, p_todo):
             except KeyError:
                 pass
 
-            p_todo_str = color + p_todo_str + NEUTRAL_COLOR
-
+            p_todo_str = '%s%s%s' % (color, p_todo_str, NEUTRAL_COLOR)
             if config().highlight_projects_contexts():
                 p_todo_str = re.sub(
                     r'\B(\+|@)(\S*\w)',
-                    PROJECT_COLOR + r'\g<0>' + color,
+                    #PROJECT_COLOR + r'\g<0>' + color,
+                    lambda m: (
+                        CONTEXT_COLOR if m.group(0)[0] == "+"
+                        else PROJECT_COLOR)+m.group(0)+color,
                     p_todo_str)
-
-            p_todo_str += NEUTRAL_COLOR
+            p_todo_str = re.sub(r'\w(\S*:[^/\s]\S*)\w',METADATA_COLOR+r'\g<0>'+color,p_todo_str)
+            # add LINK_COLOR to anything like :  (one group od charater and a :)(// and non whitespace chars)
+            p_todo_str = re.sub(r'\s(\w*:){1}(//\S*\w)',' '+LINK_COLOR+r'\1\2'+color,p_todo_str)
```

Comment:
Got it, it was 

    (^|\s)(\w*:)(//\S*\w)

found obviously on stackoverflow [python - regex line start in character set - Stack Overflow](http://stackoverflow.com/a/13199255/1435167) . Test is here on [Pythex](http://pythex.org/?regex=(%5E%7C%5Cs)(%5Cw*%3A)%7B1%7D(%2F%2F%5CS*%5Cw)&test_string=http%3A%2F%2Fexample.org%20http%3A%2F%2Fexample.org%20Lorem%20Ipsum%20Foo%20url%3Ahttp%3A%2F%2Fexample.org%20http%3A%2F%2Fexample.org&ignorecase=0&multiline=0&dotall=0&verbose=0).